src/test/java/uk/gov/justice/digital/zone/StructuredZoneTest.java

```diff
@@ -0,0 +1,189 @@
+package uk.gov.justice.digital.zone;
+
+
+import lombok.val;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.digital.config.BaseSparkTest;
+import uk.gov.justice.digital.config.JobParameters;
+import uk.gov.justice.digital.domain.model.SourceReference;
+import uk.gov.justice.digital.service.DataStorageService;
+import uk.gov.justice.digital.service.SourceReferenceService;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import java.util.*;
+import static org.apache.spark.sql.types.DataTypes.StringType;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class StructuredZoneTest extends BaseSparkTest {
+
+    private static final String S3_PATH_KEY = "dpr.structured.s3.path";
+    private static final String S3_PATH = "s3://loadjob/structured";
+
+    private final JobParameters jobParameters = new JobParameters(Collections.singletonMap(S3_PATH_KEY, S3_PATH));
+    private final DataStorageService storage = new DataStorageService();
+
+    @Mock
+    private Dataset<Row> mockedDataSet;
+
+    @Mock
+    private Optional<SourceReference> optionalRef;
```

Comment:
No need to mock java `Optional`. Just have your mock return an instance of Optional that's empty or populated as required.